### PR TITLE
0.10.0 - palig

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@rgommers / @hroncok / @jeanlaroche / @h-vetinari this is the release candidate for 0.10.0, if you have time to give it a try, that's great!

**Comment by jeanlaroche** [comment]
oh awesome!
I've started the process of adding a daily unit test to build all my 
algos with the latest master.

**Comment by h-vetinari** [comment]
So I tested this with minimal changes in https://github.com/conda-forge/pythran-feedstock/pull/56, and linux/windows are passing**, but osx gets:
```
import: 'omp'
Traceback (most recent call last):
  File "/Users/runner/miniforge3/conda-bld/pythran_1631178274877/test_tmp/run_test.py", line 2, in <module>
    import omp
  File "/Users/runner/miniforge3/conda-bld/pythran_1631178274877/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/omp/__init__.py", line 140, in <module>
    sys.modules[__name__] = OpenMP()
  File "/Users/runner/miniforge3/conda-bld/pythran_1631178274877/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/omp/__init__.py", line 33, in __init__
    self.init_not_msvc()
  File "/Users/runner/miniforge3/conda-bld/pythran_1631178274877/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.7/site-packages/omp/__init__.py", line 125, in init_not_msvc
    "or adjust the {} environment variable.".format(env_vars[0]))
ImportError: I can't find a shared library for libomp, you may need to install it or adjust the DYLD_LIBRARY_PATH environment variable.
```

CC @isuruf

** well, windows fails after cleaning up the xsimd unvendoring, but would pass if that part of the recipe was left unchanged.

**Comment by hroncok** [comment]
Trying a Fedora rawhide build.

**Comment by hroncok** [comment]
ppc64le fails with:

```
/usr/include/boost/math/tools/promotion.hpp:138:10: error: static assertion failed: Sorry, but this platform does not have sufficient long double support for the special functions to be reliably implemented.
  138 |          BOOST_STATIC_ASSERT_MSG((0 == std::is_same<type, long double>::value), "Sorry, but this platform does not have sufficient long double support for the special functions to be reliably implemented.");
      |          ^~~~~~~~~~~~~~~~~~~~~~~
```

See https://koji.fedoraproject.org/koji/taskinfo?taskID=75399643 -- https://kojipkgs.fedoraproject.org//work/tasks/9643/75399643/build.log

**Comment by serge-sans-paille** [comment]
@h-vetinari : exactly the kind of situation I didn't want to endup with :-)
@hroncok : ok, I'll disable that test for that architecture

**Comment by serge-sans-paille** [comment]
@h-vetinari / @hroncok can you give this new version another try?

**Comment by h-vetinari** [comment]
> 
> 
> @h-vetinari / @hroncok can you give this new version another try?

Same result...

**Comment by hroncok** [comment]
Started a build in https://koji.fedoraproject.org/koji/taskinfo?taskID=75444551 but I have to go offline.

**Comment by serge-sans-paille** [comment]
@hroncok thanks! There was a stupid typo in one of the test, you'll need to rerun :-/
@h-vetinari. Another blind attempt at fixing omp detection

**Comment by h-vetinari** [comment]
Still the same error, although `llvm-openmp` is definitely in the environment.

@isuruf, could you chime in here?

File: omp/__init__.py [hunk 1]
```diff
@@ -44,28 +44,8 @@ def init_msvc(self, ver):
             self.version = 20
 
     def get_libomp_names(self):
-        """Return list of OpenMP libraries to try, based on platform and
-        compiler detected."""
-        if cxx is None:
-            # Can't tell what compiler we're using, guessing we need libgomp
-            names = ['libgomp']
```
Comment (h-vetinari) [inline omp/__init__.py:51]: You could keep the changes here (assuming they had some reason in the first place) and just strip the `lib`...?

File: omp/__init__.py [hunk 2]
```diff
@@ -81,38 +61,43 @@ def init_not_msvc(self):
...
         for libomp_name in libomp_names:
-            if cxx is None or sys.platform == 'win32':
-                # Note: Clang supports -print-file-name, but not yet for
-                # clang-cl as of v12.0.0 (April '21)
-                continue
 
-            cmd = [cxx, '-print-file-name=' + libomp_name]
-            # the subprocess can fail in various ways in that case just give up
+            cmd = [cxx, '-print-file-name=' + libomp_name +
+                   self.get_shared_lib_extension()]
```
Comment (h-vetinari) [inline omp/__init__.py:74]: so this invocation does not contain the `lib` prefix but _does_  contain the extension, which seems like the neither-nor case in terms of what's a useful variant...?

File: omp/__init__.py [hunk 3]
```diff
@@ -81,38 +61,43 @@ def init_not_msvc(self):
...
-                    break
-                path = path.strip()
-                if os.path.isfile(os.path.join(path, libomp_name)):
-                    libomp_path = os.path.join(path, libomp_name)
-
+            if not libomp_path:
+                for path in paths:
+                    candidate_path = os.path.join(
+                        path,
+                        libomp_name + self.get_shared_lib_extension())
```
Comment (h-vetinari) [inline omp/__init__.py:94]: Same here

Comment (serge-sans-paille) [reply omp/__init__.py:94]: hell, fixing that

File: omp/__init__.py (hunk 1 above)
Comment (serge-sans-paille) [reply omp/__init__.py:51]: No, that's overly complex for not much gain.

**Comment by h-vetinari** [comment]
This is now working for the osx conda-forge builds. 👍

**Comment by h-vetinari** [comment]
Well, actually... this new setup is extremely noisy on windows...

<details>

```
(%PREFIX%) %SRC_DIR%>pythran -v simple_numexpr.py -DUSE_XSIMD -march=native
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
INFO:      sys file exists: D:\bld\pythran_1631293007498\_test_env\lib\site-packages\pythran\pythran.cfg
INFO: platform file exists: D:\bld\pythran_1631293007498\_test_env\lib\site-packages\pythran\pythran-win32.cfg
WARNING:     user file does not exist: C:\Users\VssAdministrator\.pythranrc
INFO: pythranrc section [pythran] is valid and options are correct
INFO: pythranrc section [typing] is valid and options are correct
INFO: pythranrc section [compiler] is valid and options are correct
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libgomp.dll' [-Wunknown-argument]
clang-cl: error: no input files
clang-cl: warning: unknown argument ignored in clang-cl: '-print-file-name=libiomp5.dll' [-Wunknown-argument]
clang-cl: error: no input files
WARNING: Failed to find 'pythran-openblas' package. Please install it or change the compiler.blas setting. Defaulting to 'blas'
running build_ext
```

</details>

**Comment by serge-sans-paille** [comment]
@h-vetinari fix pushed that should prevent this verbose stuff (?)

**Comment by h-vetinari** [comment]
> 
> 
> @h-vetinari fix pushed that should prevent this verbose stuff (?)

will try. could you comment on the `pythran.cfg` stuff in the other PR? 🙃

**Comment by h-vetinari** [comment]
Still very noisy (on windows):
```
pythran -v simple_numexpr.py -DUSE_XSIMD -march=native 
clang-cl: error: no input files
clang-cl: error: no input files
clang-cl: error: no input files
clang-cl: error: no input files
clang-cl: error: no input files
[continues]
```

**Comment by serge-sans-paille** [comment]
@h-vetinari sorry for the iteration and thanks for your help. stderr redirected to devnull, that should do the trick

**Comment by h-vetinari** [comment]
> 
> 
> @h-vetinari sorry for the iteration and thanks for your help. stderr redirected to devnull, that should do the trick

That _did_ do the trick 🥳

**Comment by serge-sans-paille** [comment]
@h-vetinari @jeanlaroche @rgommers @hroncok : thanks to your feedback, this should be good now. I'll leave one more day if you want to run a final check with the latest update, then I'll tag and release!

**Comment by hroncok** [comment]
On it: https://koji.fedoraproject.org/koji/taskinfo?taskID=75592857

**Comment by h-vetinari** [comment]
I ran another build in https://github.com/conda-forge/pythran-feedstock/pull/56 yesterday and there were no issues.

**Comment by serge-sans-paille** [comment]
Demat d'an holl

**Pull request merged**